valkeyotel/trace.go

```diff
@@ -504,8 +506,8 @@ func (d *dedicated) end(span trace.Span, err error) {
 	endSpan(span, err)
 }
 
-func startSpan(tracer trace.Tracer, ctx context.Context, op string, size int, attrs trace.SpanStartEventOption) (context.Context, trace.Span) {
-	return tracer.Start(ctx, op, kind, attr(op, size), attrs)
+func startSpan(tracer trace.Tracer, ctx context.Context, op string, size int, attrs []trace.SpanStartOption) (context.Context, trace.Span) {
+	return tracer.Start(ctx, op, append(attrs, kind, attr(op, size))...)
```

Comment:
There are a couple of places where I've changed `SpanStartEventOption` to `SpanStartOption` so that when we can play nice with `kind` when we get down into here.